File: functions/GenerateImageLabels.py
import boto3
import uuid
import json

rekognition = boto3.client("rekognition", "us-east-1")
dynamo = boto3.resource("dynamodb", "us-east-1")

max_labels = 3
min_confidence = 80


def lambda_handler(event, context):
    image_key = event["Records"][0]["s3"]["object"]["key"]
    
    response = rekognition.detect_labels(
        Image = {
            "S3Object": {
                "Bucket": "imgurclone",
                "Name": image_key,
            }
        },
        MaxLabels = max_labels,
        MinConfidence = min_confidence
    )

    labels = []
    
    for label in response["Labels"]:
        labels.append(label["Name"])
    
    table = dynamo.Table("imgur_clone_images")

    table.put_item(
        Item = {
            "key": image_key,
            "labels": labels
        }
    )

    response = {
        "statusCode": 200,
        "body": "success"
    }

    return response

File: functions/generateImageLabels.go
package main

import (
	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-lambda-go/lambda"
	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/rekognition"
)

func Handler(event events.S3Event) error {
	session, _ := session.NewSession(&aws.Config{
		Region: aws.String("us-east-1")},
	)

	key := event.Records[0].S3.Object.Key

	client := rekognition.New(session)

	result, _ := client.DetectLabels(&rekognition.DetectLabelsInput{
		Image: &rekognition.Image{
			S3Object: &rekognition.S3Object{
				Bucket: aws.String("imgurclone"),
				Name:   aws.String(key),
			},
		},
		MaxLabels:     aws.Int64(3),
		MinConfidence: aws.Float64(80),
	})

	var labels []*string

	for _, label := range result.Labels {
		labels = append(labels, label.Name)
	}

	return nil

}

func main() {
	lambda.Start(Handler)
}
